### ISSUE TITLE
support configurable compute quotas

### ISSUE THREAD

**Issue by bodenr** [softlayer/jumpgate#71]
Right now the jumpgate code has hardcoded quota-sets. See: https://github.com/softlayer/jumpgate/blob/master/jumpgate/compute/drivers/sl/quota_sets.py#L4

While I understand these defaults taken mimic those of nova's default quotas, in reality we need a more flexible means to set these quotas in jumpgate.

While it's entirely possible to (enhance) support setting quotas dynamically, jumpgates stateless nature does not fit well into a dynamic API model for setting these.

Therefore I'm thinking an initial approach may be to support setting the quotas in the jumpgate.conf file and use them in the quota response.

For example in the conf

``` ini
[compute]
quota_sets:instances = 100
quota_sets:cores = 1000
# etc
```

One could imagine a more elaborate scheme which is fine. The point here is that the hard-coded static constants aren't very flexible and limit consumption by clients.

**Comment by Neetuj** [comment]
I will work on it.  Basically move the hardcoded data from quota_sets.py to config file right?

**Comment by bodenr** [comment]
@Neetuj -- That seems like the lowest hanging fruit for this impl given jumpgate does not have persistence... It might be worth considering if we can do this per tenant like openstack supports... maybe something like:

``` ini
[compute]
quota_sets:<tenant_id>:instances = 100
quota_sets:<tenant_id>:cores = 1000
```

where `<tenant_id>` is the tenant for the quota... if we went this route probable also need to consider a way to set a global default quota set to apply to all tenants not explictly defined.

**Comment by Neetuj** [comment]
@bodenr  did i get this correct :- 
basically we want 
1) to move this hard coded data to config files jumpgate.conf
2) this new data in jumpgate.conf  and the keys should include the tenant-id information. Some function in OSQuotaSetsV2  should set the quota values in config 
3) when OSQuotaSetsV2 on_get  is called  the quota setting for that specific tenant should be returned.

if so then i have some more questions:-
 where will the function to set config values be called from ?
 Will all tenant-id quota sets be set to same values(given below) to start with? (and then cusotmized by user later?)
           "cores": 200,
            "floating_ips": 100,
            "injected_file_content_bytes": 10240,
            "injected_file_path_bytes": 255,
            "injected_files": 5,
            "instances": 10,
            "key_pairs": 100,
            "metadata_items": 128,
            "ram": 512000,
            "security_group_rules": 20,
            "security_groups": 10

**Comment by sudorandom** [comment]
`where will the function to set config values be called from ?`
I don't understand this question.

The config values will just be set in the config. on_get will read from the config for the tenant specific limits.

`Will all tenant-id quota sets be set to same values(given below) to start with? (and then customized by user later?)`

Jumpgate has no database

**Comment by sudorandom** [comment]
Closed via #88. This doesn't yet have tenant-specific quotas defined in the config. That can be handled in another issue if it's needed.
